Live at [web4bio.github.io/webgen](https://web4bio.github.io/webgen)!
Conference call via [google meet](https://sbubmi.page.link/vip) Fridays at 11AM EST

# [Web Genomics](https://www.stonybrook.edu/commcms/vertically-integrated-projects/teams/webgen_~_web_genomics/)
Precision Medicine requires an ever growing Biomedical BigData reference. This growth is particularly striking for pathologies with a genomic basis, like Cancer. Accordingly, NIH/NCI has spearheaded the establishment of interoperable multi-petabyte sized Genomic Data Commons ([gdc.cancer.gov](https://gdc.cancer.gov), also article [PMC5683428](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5683428/)) as a data ecosystem that the scientific community, including the citizen scientist, can populate with analytical applications. The goal of this project is to explore this new data-rich landscape by developing nimble Web Applications (Apps) that help find novel associations between genomic features and cancer phenotypes. For more information see [Stony Brook University VIP page](https://www.stonybrook.edu/commcms/vertically-integrated-projects/teams/_team_page/team_page.php?team=WebGen%20(Web%20Genomics)).

## To Do

1. Create a Github account, request access to this project, list your id with your name below.
2. Create a WebGen doc at https://beta.observablehq.com (use your Github credentials). Publish it.
3. Have a look at [freecodecamp.org](https://www.freecodecamp.org) as reference for JS learning.
4. Start checking your fellow participants' notebooks and Github activity.
5. Study [GDC](https://docs.gdc.cancer.gov/API/Users_Guide/Getting_Started/) (Genomics Data Commons) [API](https://docs.gdc.cancer.gov/API/Users_Guide/Getting_Started/) and note how easy it is to retrieve data. So the next step is to start doing just that in the middle of your own notes. Have a [look at mine](https://beta.observablehq.com/@jonasalmeida/webgen).
6. Review official GDC tools (https://gdc.cancer.gov/access-data/gdc-community-tools)
7. Check out some existing work (https://github.com/oncojs)

## Participants

* [Jonas Almeida](http://www.github.com/jonasalmeida), [notebook](https://beta.observablehq.com/@jonasalmeida/webgen). [firebrowse](https://github.com/episphere/firebrowse)
* [Richard Moffitt](https://github.com/rmoffitt), [sandbox notebook](https://beta.observablehq.com/@rmoffitt/webgen), [project page](/richard/richard.html), [clean notebook](https://observablehq.com/@rmoffitt/tcgaplots)
* [Fenghsi Yu](http://www.github.com/fenghsi), [notebook](https://observablehq.com/@fenghsi/webgen/2),[fenghsi.yu@stonybrook.edu]
* [Wen Cheng](http://www.github.com/wencheng921), [notebook](https://beta.observablehq.com/@wencheng921/webgen)
* [Anthony Xiang](http://www.github.com/axian0420), [notebook](https://beta.observablehq.com/@axian0420/webgen/4), [anthony.xiang@stonybrook.edu]
* [Hunter Jimenez](http://www.github.com/hunterjimenezsbu), [notebook](https://beta.observablehq.com/@huntsmna817/webgen)
* [Carolyn Bremer](http://www.github.com/cgbrem), [notebook](https://beta.observablehq.com/@cgbrem/web-genomics), [carolyn.bremer@stonybrook.edu]
* [Erika Nemeth](http://www.github.com/enemeth19), [notebook](https://beta.observablehq.com/@enemeth19/webgen) [erika.nemeth@stonybrook.edu]
* [Sam Wang](http://www.github.com/SuperZam), [notebook](https://observablehq.com/@superzam/web-genomics/5),[sammy.wang@stonybrook.edu]
* [Anuki Liyanage](https://github.com/ANUKI11),[notebook](https://observablehq.com/@anuki11/webgen/2)
* [Kevin Chang](https://github.com/Xnonx14),[notebook](https://beta.observablehq.com/@xnonx14/kevin-changs-notebook),[kevin.chang.1@stonybrook.edu]
* [Jonathan Krog](https://github.com/Jonathan-Krog), [notebook](https://beta.observablehq.com/@jonathan-krog/webgen-fall-2019)
* [Chantelle Dsilva](http://www.github.com/chantelle-dsilva), [notebook](https://beta.observablehq.com/@chantelle-dsilva/webgen), [chantelleeliza.dsilva]
* [Zhenghao (Kevin) Zhu](https://github.com/ZhenghaoZhu), [notebook](https://observablehq.com/@zhenghaozhu/webgen), [zhenghao.zhu@stonybrook.edu]

#### Alumni
  * [Tony Jin](https://github.com/t5n), [notebook](https://beta.observablehq.com/@t5n)
  * [Chinonso Nwabueze](http://www.github.com/conwabueze)

## APIs
* GDC https://api.gdc.cancer.gov
  * [documentation](https://docs.gdc.cancer.gov/API/Users_Guide/Getting_Started/#api-endpoints)
  * [example](https://api.gdc.cancer.gov/projects?fields=disease_type&size=2)
* firebrowse http://firebrowse.org
  * [documentation](http://firebrowse.org/api-docs/)
  * [example](http://firebrowse.org/api/v1/Samples/mRNASeq?format=json&gene=GATA6&cohort=PAAD&page=1&page_size=2&sort_by=tcga_participant_barcode)
* All of us https://public.api.researchallofus.org
  * [documentation] ?
  * [example] ?
____

## Course information
Participation in this project may count for SBU students as part of the [VIP program](https://www.stonybrook.edu/commcms/vertically-integrated-projects/prospective_students/how_credits_count), or [BMI 530](https://bmi.stonybrookmedicine.edu/courses) Software Development in Biomedical Informatics, or under another listing in your home department.

Issues: Cancer Genomics, Web Technologies, JavaScript, Artificial Intelligence, scalable distributed computing, social coding, patient-centered cancer prevention and treatment.

Methods & Technologies: Web Technologies (JavaScript, Apps), Cancer Genomics, Computational Statistics, Artificial Intelligence, Precision Medicine

Disciplines: Applied Mathematics and Statistics, Biomedical Engineering, Biomedical Informatics, Computer Engineering, Computer Science

Interests / Preparation by Major: Pre-existing knowledge of programming, particularly JavaScript, will be favored as it indicates an innate interest in social coding (user-facing application development). This is applicable to all students, regardless of focus being on Mathematics or Molecular Biology.

Faculty: 
    Richard Moffitt (Department of Biomedical Informatics)
    Jonas S Almeida (Department of Biomedical Informatics)

Team Established: Spring 2019

Meeting Time: Friday 11:00-11:53am

Meeting Location: HSCENT 3-045B

Contact: richard . moffitt @ stonybrookmedicine . edu

## Links
* Related work
  * [CbioPortal](https://www.cbioportal.org/)
  * [Regulome explorer](http://explorer.cancerregulome.org/)
  * [IntOGen](www.intogen.org/)
  * [Xena](genome-cancer.ucsc.edu)
  * [TCGA Splicing Variation Database](http://tsvdb.com/) [github](https://github.com/wenjie1991/TSVdb)
* Active friends of the project
  * SBU's Integrative Informatics workshop [wiki](https://sites.google.com/mathbiol.org/integrativeinformatics/home)
  * NCI Division of Cancer Epidemiology and Genetics
    * cloud4bio [github](https://github.com/cloud4bio/cloud4bio.github.com) and [wiki](https://sites.google.com/mathbiol.org/cloud4bio/home)
    * episphere [github](https://github.com/episphere) and [observable](https://observablehq.com/@episphere)
* Previous related work
  * https://github.com/mathbiol/gdc
  * https://github.com/mathbiol/gdcfun
  * https://beta.observablehq.com/@youdymoo/gdcfun
  * http://bit.ly/tcgascope
  * http://bit.ly/loadsparcs
  * http://bit.ly/pqiSuffolk


____

Edit this page at [source](https://github.com/web4bio/webgen/blob/master/README.md).
